internal/k8sinternal/client_test.go

```diff
@@ -1,12 +1,15 @@
-package k8sinternal
+package k8sinternal_test
```

Comment:
Is this the normal convention that the package test have a different name than the package itself? I've not done this before and I'm wondering if I should have been...

Reply:
I remember is something to do with blackbox/whitebox testing 💭. I think when we call this a different name, we would have to import `k8sinternal` in order to get access to the rest of the code.

Reply:
Exactly what Brian said! It depends what you want to test.

If you use the same package as the code you're testing (whitebox), you have access to the private methods in the package and can test those.

If you use the special `_test` package (normally Go won't let you have multiple package names in the same directory, so this test package is a special case), you can test your code as if you were a consumer of the package (blackbox).

In this case we're not testing any private methods so using the "test" package is kind of a cheat to get out of import cycle problems. Alternatively we could move the functions that cause the import cycle into a new package and import that from both `k8sinternal` and `test`.